modules/ui/rapid_internal_features_dialog.js

```diff
@@ -0,0 +1,153 @@
+import { t } from '../util/locale';
+import { icon } from './intro/helper';
+import { uiModal } from './modal';
+import { select as d3_select } from 'd3-selection'; 
+import { rapidInternalFeaturesStorage } from './rapid_internal_features_storage'; 
+
+
+export function uiRapidInternalFeaturesDialog(context) {
+    var featureEnabled = rapidInternalFeaturesStorage(); 
+    var aiFeatureHaloEnabled = featureEnabled('aiFeatureHalo');
+
+
+    function toggleAiFeatureHalo () {
+        var osmLayer = d3_select('.data-layer.osm'); 
+
+        aiFeatureHaloEnabled = !aiFeatureHaloEnabled; 
```

Comment:
Good catch, I think the difference in code is vestigial. I originally did the initial read of the settings as well as the toggling of the settings from this one file- I thought better of it and split them into separate files. The original bug (and the reason for this closure reference) was to read the setting before we painted the map upon startup of RapiD- otherwise, the setting wouldn't get read until the user had opened the settings dialog. *Not* ideal, hence the difference. 

TL;DR: it shouldn't have to be like this. I will feex.